src/frontend/interface/index.js

```diff
@@ -0,0 +1,66 @@
+define(['jquery', 'backend'], function(){
+
+    var continueExecution = false;
+    var executionDelay = 1000;
+
+    function nextStep(){
+        backend.nextStep()
+            .then(function(executionResult) {
+                alert(executionResult.description);
+            })
+            .catch(function(err) {
+                alert(err.stack);
+            });
+    }
+
+    function nextStepOver(){
+        backend.nextStepOver()
+            .then(function(executionResult) {
+                alert(executionResult.description);
+            })
+            .catch(function(err) {
+                alert(err.stack);
+            });
+    }
+
+    function executeLoop(){
+        nextStep();
+        if (continueExecution)
+            timeout = setTimeout(executeLoop, executionDelay);
+    }
+
+    function initiateExecution(){
+        continueExecution = true;
+        timeout = setTimeout(executeLoop, executionDelay);
+    }
+
+    $('#btn-start').click(function(){
+        backend.runProgram("x=2")
+            .then(function(result) {
+                alert("Compilation result:" + result);
+                initiateExecution();
+            })
+            .catch(function(err) {
+                alert(err.stack);
+            });
+    });
+
+    $('#btn-step').click(function(){
+        nextStep();
+    });
+
+    $('#btn-step-over').click(function(){
+        nextStepOver();
+    });
+
+    $('#btn-stop').click(function(){
+        continueExecution = false;
+    });
+
+    $('#btn-end').click(function(){
+        backend.clean()
```

Comment:
Please end all promise chains with a "done()" unless passing a promise for another method to consume